libs/elasticsearch/langchain_elasticsearch/_utilities.py

```diff
@@ -29,15 +35,66 @@ def model_must_be_deployed(client: Elasticsearch, model_id: str) -> None:
     try:
         dummy = {"x": "y"}
         client.ml.infer_trained_model(model_id=model_id, docs=[dummy])
-    except NotFoundError as err:
+    except exceptions.NotFoundError as err:
         raise err
-    except ConflictError as err:
-        raise NotFoundError(
+    except exceptions.ConflictError as err:
+        raise exceptions.NotFoundError(
             f"model '{model_id}' not found, please deploy it first",
             meta=err.meta,
             body=err.body,
         ) from err
-    except BadRequestError:
+    except exceptions.BadRequestError:
         # This error is expected because we do not know the expected document
         # shape and just use a dummy doc above.
         pass
+
+
+def setup_connection(
+    es_connection: Optional[Elasticsearch] = None,
+    es_url: Optional[str] = None,
+    es_cloud_id: Optional[str] = None,
+    es_user: Optional[str] = None,
+    es_api_key: Optional[str] = None,
+    es_password: Optional[str] = None,
+    es_params: Optional[Dict[str, Any]] = None,
+) -> Elasticsearch:
+    if es_connection is not None:
+        _es_client = es_connection
+        if not _es_client.ping():
+            raise exceptions.ConnectionError("Elasticsearch cluster is not available")
+    elif es_url is not None or es_cloud_id is not None:
+        try:
+            _es_client = create_elasticsearch_client(
+                url=es_url,
+                cloud_id=es_cloud_id,
+                api_key=es_api_key,
+                username=es_user,
+                password=es_password,
+                params=es_params,
+            )
+        except Exception as err:
+            logger.error(f"Error connecting to Elasticsearch: {err}")
+            raise err
+    else:
+        raise ValueError(
+            """Either provide a pre-existing Elasticsearch connection, \
+            or valid credentials for creating a new connection."""
+        )
+
+    return _es_client
+
+
+def manage_cache_index(
```

Comment:
since this is specific to the `cache` module, this could also be a private function there.

Reply:
yep